pkg/controller/volume/persistentvolume/scheduler_assume_cache.go

```diff
@@ -156,33 +160,42 @@ func (c *assumeCache) add(obj interface{}) {
 		return
 	}
 
-	c.rwMutex.Lock()
-	defer c.rwMutex.Unlock()
-
-	if objInfo, _ := c.getObjInfo(name); objInfo != nil {
-		newVersion, err := c.getObjVersion(name, obj)
-		if err != nil {
-			glog.Errorf("add: couldn't get object version: %v", err)
-			return
+	updated := false
+	func() {
```

Comment:
Hmm, I'm not so happy with the structural changes caused by the broadcaster.
I see you are worried that the Broadcast can take long and make the cache blocked.
Maybe we can keep the code as is, and move the Broadcast() into go routine?